### ISSUE TITLE
Tooltips in Chrome & Safari cause flicker and re-layout

### ISSUE THREAD

**Issue by brentd** [koekeishiya/yabai#424]
I apologize if this has been covered elsewhere, but I haven't been able to find or come up with a solution yet.

When hovering a link that shows a tooltip in Chrome/Safari, on a space using `yabai -m config layout bsp`, yabai seems to be trying to fit the tooltip in a new split, causing flicker.

* `yabai-v2.3.0`
* I have a [very simple config file here](https://gist.github.com/brentd/81987b5d21d1814c9133de7886c620a2)
* I tried adding `yabai -m rule --add title="^$" manage=off` hoping it would catch the (presumably unnamed) window, but no
* I tried running `yabai -m query --windows --space 1` to catch any info at all about the tooltip window but have been unsuccessful :( attaching the query of just the two windows I have open though
* I have not disabled SIP, hoping I could get by without the advanced features, could have something to do with it?

![flicker](https://user-images.githubusercontent.com/1419/75121917-bd409280-5666-11ea-8541-cde38c7f7588.gif)

```
yabai -m query --windows --space 1
[{
	"id":17925,
	"pid":34814,
	"app":"iTerm2",
	"title":"~",
	"frame":{
		"x":13.0000,
		"y":35.0000,
		"w":1260.0000,
		"h":1392.0000
	},
	"level":0,
	"role":"AXWindow",
	"subrole":"AXStandardWindow",
	"movable":1,
	"resizable":1,
	"display":1,
	"space":1,
	"visible":1,
	"focused":1,
	"split":"vertical",
	"floating":0,
	"sticky":0,
	"topmost":0,
	"border":1,
	"shadow":1,
	"zoom-parent":0,
	"zoom-fullscreen":0,
	"native-fullscreen":0
},{
	"id":17088,
	"pid":59178,
	"app":"Google Chrome",
	"title":"koekeishiya/yabai: A tiling window manager for macOS based on binary space partitioning - Google Chrome",
	"frame":{
		"x":1287.0000,
		"y":35.0000,
		"w":1260.0000,
		"h":1392.0000
	},
	"level":0,
	"role":"AXWindow",
	"subrole":"AXStandardWindow",
	"movable":1,
	"resizable":1,
	"display":1,
	"space":1,
	"visible":1,
	"focused":0,
	"split":"vertical",
	"floating":0,
	"sticky":0,
	"topmost":0,
	"border":1,
	"shadow":1,
	"zoom-parent":0,
	"zoom-fullscreen":0,
	"native-fullscreen":0
}]
```

**Comment by brentd** [comment]
Further testing seems to suggest that these two lines in my config is the culprit:

```
yabai -m rule --add app=".*" manage=off
yabai -m rule --add app="^(Google Chrome|Safari|Brave|Slack|iTerm2|Finder)$" manage=on
```

More to the point: marking everything as `manage=off`, followed by specifically allowing some apps, seems to disrupt some of the default ignoring rules yabai already sets for windows like tooltips.

Is there another way I should be saying "I only want to manage apps with these names"?

**Comment by koekeishiya** [comment]
You can have a single rule that makes everything float by default, using manage=off

Then subscribe to application_launched and window_created signal to unfloat a window that matches your criteria.

The manage=on part of the rules system is primarily meant to allow a user to force a window to be treated as a proper window, because some applications misreport their window role.

Sent from my iPhone





On 24 Feb 2020, at 04:38, Brent Dillingham <notifications@github.com> wrote:

﻿

Further testing seems to suggest that these two lines in my config is the culprit:

yabai -m rule --add app=".*" manage=off
yabai -m rule --add app="^(Google Chrome|Safari|Brave|Slack|iTerm2|Finder)$" manage=on


More to the point: marking everything as manage=off, followed by specifically allowing some apps, seems to disrupt some of the default ignoring rules yabai already sets for windows like tooltips.

Is there another way I should be saying "I only want to manage apps with these names"?

**Comment by koekeishiya** [comment]
On the master branch exclusion has been implemented (#173) and you can use the following rule:

```
yabai -m rule --add app!="^(Google Chrome|Safari|Brave|Slack|iTerm2|Finder)$" manage=off
```

**Comment by brentd** [comment]
@koekeishiya incredible! I can't wait to try it out. Thanks so much.
